CMakeLists.txt

```diff
@@ -559,12 +559,36 @@ add_custom_command(
 add_custom_target(version_h DEPENDS ${PROJECT_BINARY_DIR}/include/generated/zephyr/version.h)
 
 if(EXISTS ${APPLICATION_SOURCE_DIR}/VERSION)
+  if(CONFIG_APPLICATION_VERSION_TWEAK_FROM_GIT)
+    find_package(Git QUIET)
+    if(GIT_FOUND)
+      execute_process(
+        COMMAND ${GIT_EXECUTABLE} rev-parse HEAD
```

Comment:
Can you move this code to `cmake/gen_version_h.cmake`, where `git describe` already is? So all git commands are in the same place, otherwise it's confusing and very time-consuming to find which git command does what, when and where.

Also, it's not like the top-level `CMakeList.txt` file is too short...

Reply:
Good point, move the logic into `gen_version_h.cmake`.